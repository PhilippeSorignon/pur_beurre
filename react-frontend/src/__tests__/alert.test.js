import React from 'react';
import ReactDOM from 'react-dom';
import Alert from '../components/Alert';

import { render, cleanup } from '@testing-library/react';
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Alert renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Alert type="danger" message="Bonjour" />, div);
});

it("Alert render correctly", () => {
  const { getByRole } = render(<Alert type="danger" message="Bonjour" />);
  expect(getByRole('alert')).toHaveTextContent("Bonjour");
});

it("Alert have the right class", () => {
  const { getByRole } = render(<Alert type="danger" message="Bonjour" />);
  expect(getByRole('alert')).toHaveClass("alert-danger");
});

it("Alert matches snapshot", () => {
  const tree = renderer.create(<Alert type="danger" message="Bonjour" />).toJSON();
  expect(tree).toMatchSnapshot();
})
