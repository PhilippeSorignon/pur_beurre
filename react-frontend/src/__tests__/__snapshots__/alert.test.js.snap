// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  Bonjour
</div>
`;
